server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -120,6 +126,18 @@ public class RemoteClusterStateService implements Closeable {
 
     private final AtomicBoolean deleteStaleMetadataRunning = new AtomicBoolean(false);
 
+    private static final int CURRENT_GLOBAL_METADATA_VERSION = 1;
+
+    // TODO: For now using Gateway context, check if how we can cover all type of custom by default.
```

Comment:
If a custom type has Gateway context, that would get persisted. Which data type you are not able to cover here?

Reply:
If some custom metadata only has the Snapshot context, those can be missed. Although all custom metadata of core, are covered with Gateway, but we can't confirm for plugin or any future metadata with Snapshot context. Ideally we should to cover those as well, hence added TODO for it.

Reply:
If a custom metadata has Snapshot context only, then how is it getting persisted to disk today?

Reply:
It wont be persisted to disk as well, I think for persisting to disk as well, we use Gateway context.

Reply:
we shouldn't worry about any metadata which is not getting persisted to disk. Our reference point should be "Gateway" context only which you have done already.

Reply:
Okay, let me remove todo then. But should we highlight this somewhere, as it persisted metadata with Gateway context only?